rust/agama-autoinstall/src/cmdline.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) [2025] SUSE LLC
+//
+// All Rights Reserved.
+//
+// This program is free software; you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation; either version 2 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, contact SUSE LLC.
+//
+// To contact SUSE LLC about this file by physical or electronic mail, you may
+// find current contact information at www.suse.com.
+
+use std::{collections::HashMap, path::Path};
+
+use anyhow::Context;
+
+/// Implements a mechanism to read the kernel's command-line arguments.
+///
+/// It supports multiple values for a single key.
+pub struct CmdlineArgs(HashMap<String, Vec<String>>);
+
+impl CmdlineArgs {
```

Comment:
I really dislike name as it is confusing with CLI command line. I would call it `KernelCmdline` or call file `kernel` and then just `args` struct here...so usage will be `kernel::args`